client/lib/browser-storage/index.ts

```diff
@@ -54,15 +77,26 @@ const getDB = once( () => {
 				};
 				request.onsuccess = () => {
 					const db = request.result;
+
 					// Add a general error handler for any future requests made against this db handle.
 					// See https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Using_IndexedDB#Handling_Errors for
 					// more information on how error events bubble with IndexedDB
 					db.onerror = function( errorEvent: any ) {
+						debug( 'IDB Error', errorEvent );
 						if ( errorEvent.target?.error?.name ) {
 							mc.bumpStat( 'calypso-browser-storage', kebabCase( errorEvent.target.error.name ) );
+
+							if ( errorEvent.target.error.name === 'QuotaExceededError' ) {
```

Comment:
Should we also delete the db when this happens?

Reply:
Maybe....

These days this should only happen when we _really are_ over quota. I'm curious to see how often that happens.